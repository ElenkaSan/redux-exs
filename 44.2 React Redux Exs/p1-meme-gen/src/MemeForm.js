import React, { useState } from "react";
import { v4 as uuidv4 } from 'uuid';
import "./MemeForm";
import Button from '@mui/material/Button';
import TextField from '@mui/material/TextField';
import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid';

const DEFAULT_FORM = {
    topText: "",
    urlLink: "",
    bottomText: ""
};

function MemeForm({addingMeme}) {
    const [form, setForm] = useState(DEFAULT_FORM);

    function handleSubmit(e) {
        e.preventDefault();
        addingMeme({...form, id: uuidv4() });
        setForm(DEFAULT_FORM);
    }

    function handleChange(e) {
        const {name,value} = e.target;
        setForm(f => ({ ...f, [name]:value}));
    }

    return (
      <Grid container justifyContent="center">
        <Grid item xs={12} >
          <h2>Generate your Meme</h2>
          <Button variant="contained" color="secondary" size="large">
            <a color="success" href="https://giphy.com/reactions">
            </a> 
            Check here to find giphy image
          </Button>
          <Box
           component="form"
           onSubmit={handleSubmit}
           sx={{
             '& > :not(style)': { m: 5 },
            }}
            noValidate
            // spacing={2}
          >
          <div>
            <label htmlFor="urlLink">Image URL {' '}</label>
            <TextField 
             id="outlined-basic" 
             label="Image URL" 
             variant="outlined" 
             name="urlLink"
             onChange={handleChange}
             value={form.urlLink}
             type="text"
           /> {' '}
           <label htmlFor="topText">Top Text {' '}</label>
           <TextField 
             id="outlined-basic" 
             label="Top Text " 
             variant="outlined" 
             name="topText"
             onChange={handleChange}
             value={form.topText}
             type="text"
            /> {' '}
           <label htmlFor="bottomText">Bottom Text {' '}</label>
           <TextField 
             id="outlined-basic" 
             label="Bottom Text " 
             variant="outlined" 
             name="bottomText"
             onChange={handleChange}
             value={form.bottomText}
             type="text"
           /> {' '}
           </div>
           <div className="card">
         <Button variant="contained" color="secondary" size="large" type="submit" id="form_submit">
           Check your Meme!
         </Button>
       </div>
      </Box>
     </Grid>
    </Grid>
  )
}

export default MemeForm;