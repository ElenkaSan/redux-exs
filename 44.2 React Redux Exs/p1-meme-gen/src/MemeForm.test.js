import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
// import { render, screen, waitFor } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
import MemeForm from "./MemeForm";

let memeForm;

describe("<MemeForm /> basics", () => {
  beforeEach(() => {
    memeForm = render(<MemeForm />)
  })
  
  it('renders', () => {
    expect(memeForm).toBeTruthy();
  });
  it('matches snapshot', () => {
    expect(memeForm.asFragment()).toMatchSnapshot();
  });
})

it('updates state with topText input from form', () => {
  const mockFn = jest.fn();
  render(<MemeForm addingMeme={mockFn} />);
  fireEvent.click(screen.getByText(/Check Your Meme!/i));
  expect(mockFn).toHaveBeenCalled();
});
