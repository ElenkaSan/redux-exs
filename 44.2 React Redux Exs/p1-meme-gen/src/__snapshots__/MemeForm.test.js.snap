// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemeForm /> basics matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <h2>
        Generate your Meme
      </h2>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge css-1xnmwiw-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        <a
          color="success"
          href="https://giphy.com/reactions"
        />
        Check here to find giphy image
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <form
        class="MuiBox-root css-xnmgm1"
        novalidate=""
      >
        <div>
          <label
            for="urlLink"
          >
            Image URL  
          </label>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              Image URL
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-basic"
                name="urlLink"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Image URL
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
            
          <label
            for="topText"
          >
            Top Text  
          </label>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              Top Text 
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-basic"
                name="topText"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Top Text 
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
            
          <label
            for="bottomText"
          >
            Bottom Text  
          </label>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="outlined-basic"
              id="outlined-basic-label"
            >
              Bottom Text 
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="outlined-basic"
                name="bottomText"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-yjsfm1"
                >
                  <span>
                    Bottom Text 
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
            
        </div>
        <div
          class="card"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge css-1xnmwiw-MuiButtonBase-root-MuiButton-root"
            id="form_submit"
            tabindex="0"
            type="submit"
          >
            Check your Meme!
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
